JEXL Inspection fails to initialize under certain circumstances

<!--
SECURITY ISSUES:
Never report security issues on GitHub or other public channels (Gitter/Twitter/etc.)
Follow these instruction to report security issues: https://www.jenkins.io/security/#reporting-vulnerabilities
-->

### Version report

Jenkins and plugins versions report:

<!-- paste below the version report from https://www.jenkins.io/doc/book/system-administration/diagnosing-errors/#how-to-report-a-bug -->

```
IntelliJ IDEA 2021.1.2 (Ultimate Edition)
Build #IU-211.7442.40, built on June 1, 2021
Licensed to Denys Digtiar
Subscription is active until February 2, 2022.
Runtime version: 11.0.11+9-b1341.57 x86_64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
macOS 11.4
GC: G1 Young Generation, G1 Old Generation
Memory: 3072M
Cores: 12
Non-Bundled Plugins: IdeaVIM (0.67), Key Promoter X (2021.1.1), PsiViewer (211.6305.21-EAP-SNAPSHOT), com.intellij.ideolog (203.0.27.0), com.paperetto.dash (3.3), org.antlr.intellij.plugin (1.16), org.intellij.plugins.hcl (0.7.10), org.jetbrains.plugins.go-template (211.6693.44), mobi.hsz.idea.gitignore (4.1.0), name.kropp.intellij.makefile (211.6693.108), CheckStyle-IDEA (5.53.0), com.intellij.kubernetes (211.7442.27), zielu.gittoolbox (203.5.10), Stapler Framework Support (2.0.0), com.github.oowekyala.javacc (1.8), com.jetbrains.hackathon.indices.viewer (1.13), org.jetbrains.idea.grammar (2021.1.1), org.sonarlint.idea (4.16.0.31683), com.perl5 (2021.1), org.jetbrains.plugins.node-remote-interpreter (211.6693.65), org.jetbrains.plugins.vue (211.7142.13), org.jetbrains.kotlin (211-1.5.10-release-891-IJ7142.45), org.jetbrains.plugins.ruby (211.7442.9), Pythonid (211.7442.45), org.asciidoctor.intellij.asciidoc (0.32.52), org.jetbrains.plugins.go (211.7442.27)
Kotlin: 211-1.5.10-release-891-IJ7142.45
```

- What Operating System are you using (both controller, and any agents involved in the problem)?

```
macOS 11.4
```

### Reproduction steps

<!--
- Write bullet-point reproduction steps.
- Be explicit about any relevant configuration, jobs, build history, user accounts, etc., redacting confidential information as needed.
- The best reproduction steps start with a clean Jenkins install, perhaps a `docker run` command if possible.
- Use screenshots where appropriate, copy textual output otherwise. When in doubt, do both.
- Include relevant logs, debug if needed - https://www.jenkins.io/doc/book/system-administration/viewing-logs/
-->

- Step 1...
- Step 2...

### Results

Expected result:

JEXL inspection reports invalid expressions.

Actual result:

```
java.lang.ExceptionInInitializerError
	at org.kohsuke.stapler.idea.JexlInspection.parseJexl(JexlInspection.java:243)
	at org.kohsuke.stapler.idea.JexlInspection.check(JexlInspection.java:155)
	at org.kohsuke.stapler.idea.JexlInspection.checkXmlAttributeValue(JexlInspection.java:21)
	at org.kohsuke.stapler.idea.LocalXmlInspectionTool$1.visitXmlAttributeValue(LocalXmlInspectionTool.java:45)
	at com.intellij.psi.impl.source.xml.XmlAttributeValueImpl.accept(XmlAttributeValueImpl.java:43)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:65)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:56)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.runToolOnElements(LocalInspectionsPass.java:320)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitPriorityElementsAndInit$4(LocalInspectionsPass.java:279)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:161)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:153)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitPriorityElementsAndInit$5(LocalInspectionsPass.java:277)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:132)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:121)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitPriorityElementsAndInit$6(LocalInspectionsPass.java:277)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:92)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
	at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:83)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:178)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:83)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: org.apache.commons.logging.LogConfigurationException: org.apache.commons.logging.LogConfigurationException: org.apache.commons.logging.LogConfigurationException: Class org.apache.commons.logging.impl.Log4JLogger does not implement Log
	at org.apache.commons.logging.impl.LogFactoryImpl.newInstance(LogFactoryImpl.java:532)
	at org.apache.commons.logging.impl.LogFactoryImpl.getInstance(LogFactoryImpl.java:272)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:414)
	at org.apache.commons.jexl.ExpressionFactory.<clinit>(ExpressionFactory.java:56)
	... 32 more
Caused by: org.apache.commons.logging.LogConfigurationException: org.apache.commons.logging.LogConfigurationException: Class org.apache.commons.logging.impl.Log4JLogger does not implement Log
	at org.apache.commons.logging.impl.LogFactoryImpl.getLogConstructor(LogFactoryImpl.java:416)
	at org.apache.commons.logging.impl.LogFactoryImpl.newInstance(LogFactoryImpl.java:525)
	... 35 more
Caused by: org.apache.commons.logging.LogConfigurationException: Class org.apache.commons.logging.impl.Log4JLogger does not implement Log
	at org.apache.commons.logging.impl.LogFactoryImpl.getLogConstructor(LogFactoryImpl.java:412)
	... 36 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -16,7 +16,10 @@ repositories {
 }
 
 dependencies {
-    implementation 'org.jenkins-ci:commons-jexl:1.1-jenkins-20111212'
+    implementation('org.jenkins-ci:commons-jexl:1.1-jenkins-20111212') {
+        // Provided by the Platform
+        exclude group: "commons-logging", module: "commons-logging"
+    }
     implementation 'net.java.dev.textile-j:textile-j:2.2.864'
 
     testImplementation 'junit:junit:4.13.2'

--- a/src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java
+++ b/src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java
@@ -35,4 +35,12 @@ public class JexlInspectionTest extends LightPlatformCodeInsightFixtureTestCase 
         List<HighlightInfo> highlightInfos = myFixture.doHighlighting(HighlightSeverity.WEAK_WARNING);
         assertEmpty(highlightInfos);
     }
+
+    public void testParserFailureJexlInspection() {
+        myFixture.configureByFile(getTestName(true) + ".jelly");
+        myFixture.enableInspections(new JexlInspection());
+        List<HighlightInfo> highlightInfos = myFixture.doHighlighting(HighlightSeverity.WEAK_WARNING);
+        assertNotEmpty(highlightInfos);
+        assertEquals("Two invalid JEXL expressions are used", 2, highlightInfos.size());
+    }
 }

--- a/src/test/testData/parserFailureJexlInspection.jelly
+++ b/src/test/testData/parserFailureJexlInspection.jelly
@@ -1,0 +1,6 @@
+<j:jelly xmlns:j="jelly:core">
+    <!-- Got these from https://github.com/apache/commons-jexl/blob/c8af6a07df91d2a6d6d22adebf7c52ab02d2b74e/src/test/org/apache/commons/jexl/ParseFailuresTest.java -->
+    <j:if test="${?}">
+        ${eq}
+    </j:if>
+</j:jelly>

